generic_server.cc

```diff
@@ -129,16 +129,15 @@ future<> server::shutdown() noexcept {
         l.abort_accept();
         _logger.debug("abort accept {} out of {} done", ++nr, nr_total);
     }
-    auto nr_conn = make_lw_shared<size_t>(0);
+    size_t nr_conn = 0;
     auto nr_conn_total = _connections_list.size();
     _logger.debug("shutdown connection nr_total={}", nr_conn_total);
-    return parallel_for_each(_connections_list.begin(), _connections_list.end(), [this, nr_conn, nr_conn_total] (auto&& c) {
-        return c.shutdown().then([this, nr_conn, nr_conn_total] {
-            _logger.debug("shutdown connection {} out of {} done", ++(*nr_conn), nr_conn_total);
-        });
-    }).then([this] {
-        return std::move(_listeners_stopped);
-    });
+    co_await parallel_for_each(_connections_list.begin(), _connections_list.end(),
```

Comment:
We have a seastar::coroutine::parallel_for_each() which is slightly faster (allocates state within the outer coroutine rather than in a separate block), and is immune to the lambda coroutine fiasco. May as well use it.

Reply:
@avikivity

> We have a seastar::coroutine::parallel_for_each() which is slightly faster (allocates state within the outer coroutine rather than in a separate block), and is immune to the lambda coroutine fiasco. May as well use it.

I find the namespace usage slightly confusing. Per `CONTRIBUTING.md`, *Scylla code is written with `using namespace seastar`*, and so we have to watch for the difference between plain `parallel_for_each()` (from `seastar/core/loop.hh`) versus `coroutine::parallel_for_each()` (from `seastar/coroutine/parallel_for_each.hh`).

But yes, it does seem to work.

@bhalevy

> I think it should also be able to work directly on `_connections_list` rather than on `.begin(), .end()`

That's right; there's an overload at the end of `seastar/coroutine/parallel_for_each.hh` that takes a `Range&&`.

Thank you both for the suggestions; I've implemented these in v6.